src/Avatar/Avatar.js

```diff
@@ -44,18 +47,23 @@ export default function Avatar(props, context) {
 
   const classes = context.styleManager.render(styleSheet);
   const className = classNames(classes.root, {
-    [classes.defaultColor]: icon && !src && !srcSet,
+    [classes.defaultColor]: childrenProp && !src && !srcSet,
   }, classNameProp);
-
   const containerProps = {
     className,
     ...other,
   };
 
   let children = null;
 
-  if (icon) {
-    children = icon;
+  if (childrenProp) {
+    // If it's an icon, style it.
+    if (React.isValidElement(childrenProp)) {
```

Comment:
Thanks for your patience! - It seems like my coding ability has regressed 2 years in 6 months, :worried: